packages/optimizely-sdk/lib/utils/semantic_version/index.ts

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright 2020, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { getLogger } from '@optimizely/js-sdk-logging';
+import { VERSION_TYPE, LOG_MESSAGES } from '../enums';
+
+const MODULE_NAME = 'SEMANTIC VERSION';
+const logger = getLogger();
+
+/**
+  * Evaluate if provided string is number only
+  * @param  {unknown}  content
+  * @return {boolean}  true if the string is number only
+  *                     
+  */
+  function isNumber(content: string): boolean {
+    return content.match(/^[0-9]+$/) != null ? true : false;
```

Comment:
Is this equivalent?
```suggestion
    return /^\d+$/.test(content);
```
(see: [Regular Expression docs for JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Character_Classes))

Also: I suggest declaring regular expression once higher up in module scope, then reference here (`DIGIT_STRING_REGEXP.test(content)`).